src/core/WebGLPathTracer.js

```diff
@@ -98,6 +99,75 @@ export class WebGLPathTracer {
 
 		return Boolean( this._pathTracer.isCompiling );
 
+	}
+	//* Denoiser ----------------------------
+
+	get isDenoising() {
+
+		return this._denoiserManager.isDenoising;
+
+	}
+
+	get weightsUrl() {
+
+		return this._denoiserManager.denoiser.tzaUrl;
+
+	}
+
+	set weightsUrl( url ) {
+
+		this._denoiserManager.denoiser.weightsUrl = url;
+
+	}
```

Comment:
Instead of taking the URLs on the WebGLPathTracer object I'd prefer passing something like an "OIDNDenoiser" object to a function to assign it - similar to three.js [GLTFLoader.setDracoLoader](https://github.com/mrdoob/three.js/blob/f5eaae88a09fce661a814d2bd3526f1690302bb6/examples/webgl_loader_gltf_avif.html#L62) or the [setBVHWorker](https://github.com/gkjohnson/three-gpu-pathtracer/blob/main/example/index.js#L137) functions. Then any options can be set on that denoiser instance and it would let use swap in other denoisers in the future:

```js
const pathTracer = new WebGLPathTracer( renderer );
console.log( pathTracer.denoiser ); // null
console.log( pathTracer.enableDesnoiser ); // false

const denoiser = new OIDNDenoiser( options );
denoiser.setTZAPath( ... ); // see DracoLoader's "setDecoderPath" function
denoiser.debugging = true;
denoiser.setAux( ... );
// any other denoiser options etc

pathTracer.setDenoiser( denoiser );

// adjusting options later
pathTracer.denoiser.debugging = false; // etc
```

Reply:
oof ok I'll work on this refactor... I can use the `DenoiserManager` as the basis for this.

Reply:
Done!